File: Programmers/level1/Python/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    wins = set(win_nums)&set(lottos)
    zeroCount = lottos.count(0)
    choice = set(win_nums)-wins
    
    score = len(wins)
    maxScore = score + min(len(choice), zeroCount)
    
    maxRank = 7-maxScore if maxScore > 1 else 6
    minRank = 7-score if score > 1 else 6
    return [maxRank, minRank]

File: Programmers/level1/Swift/로또의 최고 순위와 최저 순위.swift
import Foundation

func solution(_ lottos:[Int], _ win_nums:[Int]) -> [Int] {
    var count: Int = 0
    var zeroCount: Int = 0
    for lot in lottos {
        if win_nums.contains(lot) {
            count += 1
        }
        else if(lot == 0) {
            zeroCount += 1
        }
    }
    var firstScore: Int = getScore(count+zeroCount)
    var lastScore: Int = getScore(count)
    var result: [Int] = []
    result.append(firstScore)
    result.append(lastScore)
    return result
}

func getScore(_ count: Int) -> Int {
    var score: Int = 0
    switch(count) {
        case 2:
        score = 5
        case 3:
        score = 4
        case 4:
        score = 3
        case 5:
        score = 2
        case 6:
        score = 1
        default:
        score = 6
    }
    return score
}